packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -26,7 +26,7 @@ const preloadMarkerRE = new RegExp(`"${preloadMarker}"`, 'g')
  */
 function preload(baseModule: () => Promise<{}>, deps?: string[]) {
   // @ts-ignore
-  if (!__VITE_IS_MODERN__ || !deps) {
+  if (!__VITE_IS_MODERN__ || !deps || deps.length === 0) {
```

Comment:
`deps` is always going to be a well defined array now, no?
```suggestion
  if (!__VITE_IS_MODERN__ || deps.length === 0) {
```

Reply:
Looks like this wasnt the case, sorry 👀

Reply:
The function signature says `deps?: string[]`
@patak-js Do you request to change it to `deps: string[]`? Or just revert the `if-condition`?

Reply:
Yes,because the declaration of `deps` is optional, do you suggest to revert commit or delete `?`.

Reply:
Check how this function is called, if in all places where it is called we are going to be passing an array, then yes, I think we can change it to `deps: string[]`. I thought that was the case, but maybe there is still another place were it is called with deps undefined.

Reply:
In the following two places, `deps` passes in `void 0`
https://github.com/vitejs/vite/blob/98d95e3e11bbc43e410b213b682e315b9344d2d7/packages/vite/src/node/plugins/importAnalysisBuild.ts#L159

https://github.com/vitejs/vite/blob/98d95e3e11bbc43e410b213b682e315b9344d2d7/packages/vite/src/node/plugins/importAnalysisBuild.ts#L282-L284

It is not recommended to modify these two places to'[ ]', because I don't know what kind of test cases to write to ensure the correctness of the two places, so I think the judgment of `!dep` is still necessary.